godot-core/src/builtin/array.rs

```diff
@@ -149,14 +159,29 @@ impl<T: GodotType> Array<T> {
     /// `sort_unstable`.
     #[doc(alias = "sort_custom")]
     pub fn sort_unstable_custom(&mut self, func: Callable) {
-        self.as_inner().sort_custom(func);
+        // SAFETY: We do not write any values that dont already exist in the array, so all values have the correct type.
+        unsafe { self.as_inner() }.sort_custom(func);
     }
 
     /// Shuffles the array such that the items will have a random order. This method uses the
     /// global random number generator common to methods such as `randi`. Call `randomize` to
     /// ensure that a new seed will be used each time if you want non-reproducible shuffling.
     pub fn shuffle(&mut self) {
-        self.as_inner().shuffle();
+        // SAFETY: We do not write any values that dont already exist in the array, so all values have the correct type.
+        unsafe { self.as_inner() }.shuffle();
+    }
+
+    /// Shrinks the array down to `new_size`.
+    ///
+    /// If you want to increase the size of the array, use [`resize_with`](Array::resize_with) instead.
```

Comment:
Should document that if requested size is larger than current size, nothing happens.

Reply:
I'm thinking it could be beneficial to return some indicator showing the results of the operation. Maybe not `Result`, even `bool` would be enough IMO.

Reply:
Yep, I agree. Definitely not `#[must_use]`, but a bool could be nice, and we already have it for free.
It would enable a simple pattern:
```rs
if array.shrink(len) {
    // Follow-up logic after shrinking
    ...
}
```